appcues/src/main/java/com/appcues/ui/primitive/ExperiencePrimitive.kt

```diff
@@ -0,0 +1,100 @@
+package com.appcues.ui.primitive
+
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.isSystemInDarkTheme
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.BoxScope
+import androidx.compose.material.ripple.rememberRipple
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.semantics.Role
+import coil.compose.AsyncImage
+import com.appcues.data.model.ExperiencePrimitive
+import com.appcues.data.model.ExperiencePrimitive.BoxPrimitive
+import com.appcues.data.model.ExperiencePrimitive.ButtonPrimitive
+import com.appcues.data.model.ExperiencePrimitive.EmbedHtmlPrimitive
+import com.appcues.data.model.ExperiencePrimitive.HorizontalStackPrimitive
+import com.appcues.data.model.ExperiencePrimitive.ImagePrimitive
+import com.appcues.data.model.ExperiencePrimitive.TextPrimitive
+import com.appcues.data.model.ExperiencePrimitive.VerticalStackPrimitive
+import com.appcues.data.model.styling.ComponentStyle
+import com.appcues.ui.LocalAppcuesActionDelegate
+import com.appcues.ui.LocalAppcuesActions
+import com.appcues.ui.extensions.PrimitiveGestureProperties
+import com.appcues.ui.extensions.blurHashPlaceholder
+import com.appcues.ui.extensions.getBoxAlignment
+import com.appcues.ui.extensions.getImageLoader
+import com.appcues.ui.extensions.getImageRequest
+import com.appcues.ui.extensions.innerPrimitiveStyle
+import com.appcues.ui.extensions.outerPrimitiveStyle
+import com.appcues.ui.extensions.toImageAsyncContentScale
+import com.appcues.ui.utils.rememberBlurHashDecoded
+
+@Composable
+internal fun ExperiencePrimitive.Compose() {
+    Box(
```

Comment:
every primitive has its down box that applies the outerPrimitiveStyle (getRole for each type too on line 70)

and also calls BackgroundImage on line 53 that will compose the image matchingParentSize below the primitive if backgroundImage is present in style.

Reply:
nice to centralize those outerPrimitiveStyle items!